iset.mm

```diff
@@ -99758,6 +99758,64 @@ A Cauchy sequence (as defined here, which has a rate of convergence
     LXCSVPWEXESVGURVHVPVRWAVPVQCDZVRNDABVIZVQPQVPVSVTWTXAUMVPXJTVRIJXKVQVJQVPVS
     VTWTXAVPVNTVSIJWOAVJQVPVOTVTIJWPBVJQVKVLVM $.
 
+  $( An upper bound for ` { A , B } ` .
+     (Contributed by Jim Kingdon, 20-Dec-2021.) $)
+  maxabsle $p |- ( ( A e. RR /\ B e. RR ) ->
+      A <_ ( ( ( A + B ) + ( abs ` ( A - B ) ) ) / 2 ) ) $=
+    ? $.
+
+  $( The maximum of two reals is no smaller than the first real.
+     Lemma 3.10 of [Geuvers], p. 10.
+     (Contributed by Jim Kingdon, 21-Dec-2021.) $)
+  maxle1 $p |- ( ( A e. RR /\ B e. RR ) -> A <_ sup ( { A , B } , RR , < ) ) $=
+    ? $.
+
+  $( The maximum of two reals is no smaller than the second real.
+     Lemma 3.10 of [Geuvers], p. 10.
+     (Contributed by Jim Kingdon, 21-Dec-2021.) $)
+  maxle2 $p |- ( ( A e. RR /\ B e. RR ) -> B <_ sup ( { A , B } , RR , < ) ) $=
+    ? $.
+
+  $( The maximum of two reals is commutative.
+     Lemma 3.9 of [Geuvers], p. 10.
+     (Contributed by Jim Kingdon, 21-Dec-2021.) $)
+  maxcom $p |- ( ( A e. RR /\ B e. RR ) ->
+      sup ( { A , B } , RR , < ) = sup ( { B , A } , RR , < ) ) $=
+    ? $.
+
+  $( Equivalence of ` <_ ` and being equal to the maximum of two reals.
+     Lemma 3.12 of [Geuvers], p. 10.
+     (Contributed by Jim Kingdon, 21-Dec-2021.) $)
+  maxleb $p |- ( ( A e. RR /\ B e. RR ) ->
+      ( A <_ B <-> sup ( { A , B } , RR , < ) = B ) ) $=
+    ? $.
+
+  $( Absolute value defined in terms of maximum.
+     Definition 3.13 of [Geuvers], p. 11.
+     (Contributed by Jim Kingdon, 21-Dec-2021.) $)
+  dfabsmax $p |- ( A e. RR -> ( abs ` A ) = sup ( { A , -u A } , RR , < ) ) $=
+    ? $.
+
+  ${
+    maxabslemlub.a $e |- ( ph -> A e. RR ) $.
+    maxabslemlub.b $e |- ( ph -> B e. RR ) $.
+    maxabslemlub.c $e |- ( ph -> C e. RR ) $.
+    maxabslemlub.clt $e |- ( ph ->
+      C < ( ( ( A + B ) + ( abs ` ( A - B ) ) ) / 2 ) ) $.
+
+    $( Lemma for ~ maxabs . A least upper bound for ` { A , B } ` .
+       (Contributed by Jim Kingdon, 20-Dec-2021.) $)
+    maxabslemlub $p |- ( ph -> ( C < A \/ C < B ) ) $=
+      ? $.
+  $}
```

Comment:
I don't understand the need for this lemma ? Once you have dfabsmax, you deduce
a+b+|a-b| = a+b+max(a-b, b-a) = max(2a,2b) = 2max(a,b)
I confess that this looks a bit tedious to formalize, since the proper way is probably to state it as the supremum of a translated subset in an ordered group --- or at least (RR,+,<) --- and then of a positive dilation of a subset of RR...